server/src/app.controller.ts

```diff
@@ -1,57 +1,52 @@
 import ApiFactory from './api/import';
-
+import * as Dtos from './app.dtos';
+import { BrowseService, TestService } from './app.service';
 
 function getApi(description: string) {
   return ApiFactory.getInstance().getApi('v1.0', description);
 }
 
-function processRequest(func: Function, body): string {
-  return JSON.stringify(func(body));
-}
 @Controller()
 export class AppController {
-  constructor(private readonly testService: TestService, 
-    private readonly browseService: BrowseService) {}
+  constructor(private readonly testService: TestService,
+    private readonly browseService: BrowseService) { }
 
   @Get(ApiFactory.getInstance().getApi('test', '/test1/test2/get'))
   getTest(): string {
     return this.testService.getTest();
   }
 
-  @Get(getApi('/browse/question-list/get'))
-  getBrowseQuestionList(@Body() body): string {
-    return processRequest(this.browseService.getBrowseQuestionList, body);
+  async getBrowseQuestionList(@Body() body: Dtos.getQuestionListInDto): Promise<Dtos.getQuestionListOutDto> {
+    console.log("question list input", body);
+    return this.browseService.getBrowseQuestionList(body);
   }
 
   @Get(getApi('/browse/question/get'))
-  getBrowseQuestion(@Body() body): string {
-    return processRequest(this.browseService.getBrowseQuestion, body);
+  async getBrowseQuestion(@Body() body: Dtos.getQuestionInDto): Promise<Dtos.getQuestionOutDto> {
+    console.log("question input", body)
+    return this.browseService.getBrowseQuestion(body);
   }
 
-  @Get(getApi('/browse/answer-list/get'))
-  getBrowseAnswerList(@Body() body): string {
-    return processRequest(this.browseService.getBrowseAnswerList, body);
-  }
+  // @Post("/makeQuestion")
+  // makeQuestion(@Body() myDto: MyDto): Promise<object> {
+  //   // console.log(request)
+  //   console.log(myDto)
+
+  //   return this.browseService.makeQuestion(myDto.questionText);
+  //   // return this.browseService.makeQuestion(myDto.userRef, myDto.questionText);
+  //   // return this.browseService.getBrowseQuestion(request.questionID);
+  // }
 
   @Get(getApi('/browse/answer/get'))
-  getBrowseAnswer(@Body() body): string {
-    return processRequest(this.browseService.getBrowseAnswer, body);
+  getBrowseAnswer(@Body() body: Dtos.getAnswerInDto): Promise<Dtos.getAnswerOutDto> {
+    return this.browseService.getBrowseAnswer(body);
+    // return this.browseService.getBrowseAnswer(getAnswerDTO.qid, getAnswerDTO.aids);
   }
 
   @Post(getApi('/browse/answer/post'))
-  postBrowseAnswer(@Body() body): string {
-    return processRequest(this.browseService.postBrowseAnswer, body);
-  }
-
-  @Post(getApi('/browse/response/post'))
-  postBrowseResponse(@Body() body): string {
-    return processRequest(this.browseService.postBrowseResponse, body);
-  }
-
-  @Get(getApi('/browse/response-limit/get'))
-  getBrowseResponseLimit(@Body() body): string {
-    return processRequest(this.browseService.getBrowseResponseLimit, body);
```

Comment:
What happened to all this?

Reply:
I'll add it back. I rebased to include your commit but ended up not using most of your code